mitmproxy/io/compat.py

```diff
@@ -112,8 +112,22 @@ def convert_200_300(data):
     data["version"] = (3, 0, 0)
     data["client_conn"]["mitmcert"] = None
     data["server_conn"]["tls_version"] = None
+    data["client_conn"]["tls_established"] = data["server_conn"]["ssl_established"] or None
+    data["client_conn"]["timestamp_tls_setup"] = data["server_conn"]["timestamp_ssl_setup"] or None
+    data["server_conn"]["tls_established"] = data["server_conn"]["ssl_established"] or None
+    data["server_conn"]["timestamp_tls_setup"] = data["server_conn"]["timestamp_ssl_setup"] or None
+    del data["client_conn"]["ssl_established"]
+    del data["client_conn"]["timestamp_ssl_setup"]
+    del data["server_conn"]["ssl_established"]
+    del data["server_conn"]["timestamp_ssl_setup"]
+
     if data["server_conn"]["via"]:
         data["server_conn"]["via"]["tls_version"] = None
+        data["server_conn"]["via"]["tls_established"] = data["server_conn"]["via"]["ssl_established"] or None
+        data["server_conn"]["via"]["timestamp_tls_setup"] = data["server_conn"]["via"]["timestamp_ssl_setup"] or None
+        del data["server_conn"]["via"]["ssl_established"]
+        del data["server_conn"]["via"]["timestamp_ssl_setup"]
+
```

Comment:
This should be a new flow file version in version.py so that dumps created with developer snapshots remain working. That's why we split version and flow file version. :)